zerver/tests/test_subs.py

```diff
@@ -397,6 +397,30 @@ def test_make_stream_public(self) -> None:
         self.assertFalse(stream.invite_only)
         self.assertTrue(stream.history_public_to_subscribers)
 
+    def test_make_web_public_stream(self) -> None:
+        user_profile = self.example_user('hamlet')
+        self.login_user(user_profile)
+        realm = user_profile.realm
+        self.make_stream('web_public_stream',
+                         realm=realm,
+                         is_web_public=True,
+                         history_public_to_subscribers=True)
```

Comment:
I don't understand this test; it seems like the parameters you set via the API (below) match how you created it?

Reply:
I wanted to check if a user can create a web-public stream and if it was accessible/editable to other users as well.

hmm, I guess just checking if it was created right is better?

Reply:
Yeah, exactly, I'd just do `Stream.objects.get()` if the goal is to verify creation happened successfully.

Reply:
(Is the intent to use `create_stream_if_needed`?)